maestrowf/datastructures/specifications/__init__.py

```diff
@@ -0,0 +1,65 @@
+"""Module containing datastructures for parsing specifications."""
+from importlib import import_module
+from inspect import getmodule
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SpecificationFactory(object):
+    """A factory class to retrieve different types of Specifications."""
+
+    _classes = {
+        "yaml": {
+            "latest":   (".yamlspecification", "YAMLSpecification"),
+            "1.0":      (".yamlspecification", "YAMLSpecification"),
+        },
+    }
+
+    @classmethod
+    def get_specification(cls, spec_name, version="latest"):
+        """
+        Look up and retrieve a Specification by name.
+
+        :param spec_name: Name of the Specification to find.
```

Comment:
Is `spec_name` referring to the specification format (e.g., yaml)?

Reply:
Yes and no -- I couldn't figure out a good name for a Maestro YAML specification, but that could also be something like "cwl" for Common Workflow Language and things like that. I'm definitely welcome to suggestions either for naming or a better way to handle the "naming" to make it clearer.

Reply:
How about `spec_type` or `spec_kind`?

Reply:
@dinatale2 -- I updated to `spec_type`.